isis/tests/FunctionalTestsMvic2isis.cpp

```diff
@@ -0,0 +1,489 @@
+  ASSERT_EQ(double(inst["RalphExposureDuration"]), 0.59168);
+  ASSERT_EQ(double(inst["TdiRate"]), 54.0833);
+  ASSERT_EQ(inst["StartTime"][0].toStdString(), "2007-02-28T06:00:00.520");
+
+  // Archive Group
+  PvlGroup &archive = isisLabel->findGroup("Archive", Pvl::Traverse);
+  ASSERT_EQ(archive["HighSpeedCompressionMode"][0].toStdString(), "LOSSLESS");
+  ASSERT_EQ(archive["ObservationCompletionStatus"][0].toStdString(), "COMPLETE");
+  ASSERT_EQ(archive["SequenceDescription"][0].toStdString(), "MVIC terminator flat");
+
+  // BandBin Group
+  PvlGroup &bandbin = isisLabel->findGroup("BandBin", Pvl::Traverse);
+  ASSERT_EQ(bandbin["Name"][0].toStdString(), "CH4");
+  ASSERT_EQ(double(bandbin["Center"]), 895);
+  ASSERT_EQ(double(bandbin["Width"]), 40);
+  ASSERT_EQ(double(bandbin["OriginalBand"]), 1);
+
+  // Kernels Group
+  PvlGroup &kernel = isisLabel->findGroup("Kernels", Pvl::Traverse);
+  ASSERT_EQ(int(kernel["NaifFrameCode"]), -98908);
+  ASSERT_EQ(kernel["NaifFrameCode"].unit().toStdString(), "SPICE ID");
+
+  // RadiometricCalibration Group
+  PvlGroup &radiometricCalibration = isisLabel->findGroup("RadiometricCalibration", Pvl::Traverse);
+  ASSERT_EQ(double(radiometricCalibration["PixelSize"]), 13.0000);
+  ASSERT_EQ(double(radiometricCalibration["PixelFov"]), 19.8065);
+  ASSERT_EQ(double(radiometricCalibration["Gain"]), 58.6000);
+  ASSERT_EQ(double(radiometricCalibration["ReadNoise"]), 30.0000);
+  ASSERT_EQ(double(radiometricCalibration["TdiRate"]), 54.0833);
+  ASSERT_EQ(double(radiometricCalibration["SolarSpectrumResolved"]), 0.0394);
+  ASSERT_EQ(double(radiometricCalibration["SolarSpectrumUnresolved"]), 1.55E-11);
+  ASSERT_EQ(double(radiometricCalibration["PholusSpectrumResolved"]), 0.0385);
+  ASSERT_EQ(double(radiometricCalibration["PholusSpectrumUnresolved"]), 1.51E-11);
+  ASSERT_EQ(double(radiometricCalibration["CharonSpectrumResolved"]), 0.0394);
+  ASSERT_EQ(double(radiometricCalibration["CharonSpectrumUnresolved"]), 1.54E-11);
+  ASSERT_EQ(double(radiometricCalibration["JupiterSpectrumResolved"]), 0.0474);
+  ASSERT_EQ(double(radiometricCalibration["JupiterSpectrumUnresolved"]), 1.86E-11);
+  ASSERT_EQ(double(radiometricCalibration["PlutoSpectrumResolved"]), 0.0398);
+  ASSERT_EQ(double(radiometricCalibration["PlutoSpectrumUnresolved"]), 1.56E-11);
+  ASSERT_EQ(double(radiometricCalibration["SolarPivotWavelength"]), 8.86E-5);
+  ASSERT_EQ(double(radiometricCalibration["JupiterPivotWavelength"]), 8.84E-5);
+  ASSERT_EQ(double(radiometricCalibration["PholusPivotWavelength"]), 8.87E-5);
+  ASSERT_EQ(double(radiometricCalibration["PlutoPivotWavelength"]), 8.86E-5);
+  ASSERT_EQ(double(radiometricCalibration["CharonPivotWavelength"]), 8.86E-5);
+
+  Histogram *hist = cube.histogram();
+
+  ASSERT_DOUBLE_EQ(hist->Average(), 0.34910885492960614);
+  ASSERT_DOUBLE_EQ(hist->Sum(), 1.0473265647888184);
+  ASSERT_EQ(hist->ValidPixels(), 3);
+  ASSERT_DOUBLE_EQ(hist->StandardDeviation(), 0.6046742741102703);
+
+  delete hist;
```

Comment:
probably a good place for a smart pointer (we should probably start using them more). std::unique_ptr probably work better here?